0024-pre-shared-key-v2/0024-pre-shared-key-v2.md

```diff
@@ -0,0 +1,266 @@
+
+**Malicious Connectors:** Exchange rates can also change for illegitimate reasons, such as connectors driving up the price mid-payment. Ultimately, the main force keeping connectors' rates low is competition. If there is only one path, the connectors do not need to do anything sneaky to drive up the rate. If a sender has multiple options for the connector they send through, they may want to use [Multi-Path Chunked Payments](#multi-path-chunked-payments) to automatically switch paths if one becomes more expensive.
+
+**Trusting the Receiver:** When doing any Interledger payment aside from an atomic exchange of money for goods or services, the sender must trust the receiver to deliver the goods once they receive the payment. With chunked payments, the sender must trust the receiver to deliver the goods and accurately report how much money has arrived on their incoming transfer. A malicious receiver could keep reporting less money having arrived on every transaction in an attempt to extract more money from the sender, but this is no worse a threat than a receiver demanding more money after a non-chunked payment arrives but before they release the goods or services. Since senders will judge merchants' prices by the total source amount they pay, receivers that pull such tricks should develop negative reputations and be avoided.
+
+## Model of Operation
+
+### Informational Quote
+
+1. Sender creates a packet with a type of 1, a random Payment ID, a Sequence number of 0, a Payment Amount of 0, a Chunk Amount set to 18446744073709551615 (the maximum value) and [encrypts it](#encryption-envelope).
+2. Sender generates a random 32-byte condition.
+3. Sender sends a test payment with the condition, encrypted data and either the chosen source amount (in the case of a fixed source amount payment) or an arbitrary amount (in the case of a fixed destination amount).
+4. Receiver is notified of the incoming transfer and decrypts the data.
+5. Receiver compares the Chunk Amount from the packet with the incoming transfer amount and notes that the incoming transfer amount is too low.
+6. Receiver creates a response packet, including the Payment ID and Sequence from the request, a Payment Amount of 0 (or the total they have fulfilled with this Payment ID), and the Chunk Amount set to the amount that arrived on the incoming transfer. Receiver [encrypts](#encryption-envelope) the packet.
+7. Receiver rejects incoming transfer, setting the rejection message's data to the encrypted quote response.
+8. Sender decrypts the quote response and divides the receiver's Chunk Amount by the source amount to determine the path exchange rate.
+
+### Single Payment
+
+1. In the case of a fixed source amount payment, sender multiplies the source amount by the rate to determine the Payment Amount. In the case of a fixed destination amount payment, sender divides the destination amount by the rate to determine the amount for their outgoing transfer.
```

Comment:
I would prefer to use ilqp in this case

Reply:
I don't think that will realistically be an option. As I've said before, ILQP doesn't work unless every connector supports it and/or every connector pushes up to date price information to one another very frequently. This is a high burden to place on the connectors that isn't necessary if senders can get the same information in a different way that doesn't require a specific feature from the connectors.